api4/work_templates.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/app/worktemplates"
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+func (api *API) InitWorkTemplate() {
+	api.BaseRoutes.WorkTemplates.Handle("/categories", api.APISessionRequired(areWorkTemplatesEnabled(getWorkTemplateCategories))).Methods("GET")
+	api.BaseRoutes.WorkTemplates.Handle("/categories/{category}/templates", api.APISessionRequired(areWorkTemplatesEnabled(getWorkTemplates))).Methods("GET")
+	api.BaseRoutes.WorkTemplates.Handle("/execute", api.APIHandler(areWorkTemplatesEnabled(executeWorkTemplate))).Methods("POST")
+}
+
+func areWorkTemplatesEnabled(h handlerFunc) handlerFunc {
+	return func(c *Context, w http.ResponseWriter, r *http.Request) {
+		if !c.App.Config().FeatureFlags.WorkTemplate {
+			c.App.Log().Warn("trying to access work templates api while feature flag is disabled")
```

Comment:
Please use `c.Logger.Warn`

Reply:
Thanks, I was unaware of the existence of a logger in the context!